DepotDownloader/ContentDownloader.cs

```diff
@@ -337,19 +337,54 @@ public static void ShutdownSteam3()
 
         public static async Task DownloadPubfileAsync(uint appId, ulong publishedFileId)
         {
-            var details = await steam3.GetPublishedFileDetails(appId, publishedFileId);
+            List<ValueTuple<string, string>> fileUrls = new();
+            List<ulong> contentFileIds = new();
 
-            if (!string.IsNullOrEmpty(details?.file_url))
+            var details = await steam3.GetPublishedFileDetails(appId, publishedFileId);
+            if (details.file_type == (uint)EWorkshopFileType.Collection)
             {
-                await DownloadWebFile(appId, details.filename, details.file_url);
+                foreach (var child in details.children)
+                {
+                    var childDetails = await steam3.GetPublishedFileDetails(appId, child.publishedfileid);
+                    if (!string.IsNullOrEmpty(childDetails?.file_url))
```

Comment:
can you turn processing a file into a reused function (which will make it recurse by default)?

Reply:
My first pass on it did that, but it turned out to require a larger refactor since DownloadAppAsync wasn't designed to be able to be called more than once per execution. It crashed when attempting to initialize the config a second time. I didn't dig in too far to see what other pieces may be unhappy getting hit more than once

Reply:
I think you can make a separate method that just processes the item itself and the fileUrls/contentFileIds are passed into as params?

Reply:
I misunderstood the first time, but good idea. I think that's cleaner. Does this look alright? (Seems to still work fine)